### PR TITLE
[Fix] Make supervisord manage the cfn-hup program in no-daemon mode

### PR DESCRIPTION
**Pull request by gmarciani**
### Description of changes
Make supervisord manage the cfn-hup program in no-daemon mode to avoid systemic failures when the program gets restarted.

**Note** the 120 second period is the interval that we currently use when running cfn-hup in daemon mode, as per [this configuration](https://github.com/aws/aws-parallelcluster/blob/develop/cli/src/pcluster/templates/cluster_stack.py#L1162-L1162)

### Tests
* Integration Tests: `test_update_slurm`
* Verified that cfn-hup is correctly restarted by supervisord when killed
* Verified that cfn-hup is correctly terminated when supervisord is stopped
* Verified that cfn-hup is correctly restarted when supervisord is restarted
* Verified that only one cfn-hup process is running at any given time
* Verified that no cfn-hup process is running when supervisord is stopped.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -30,6 +30,9 @@ This file is used to list changes made in each version of the AWS ParallelCluste
 **BUG FIXES**
 - Fix IP association on instances with multiple network cards.
 
+**BUG FIXES**
+- Fix issue causing cfn-hup daemon to fail when it gets restarted.
```
Comment (enrico-usai) [inline CHANGELOG.md:34]: Which kind of failures are we seeing? Is there a bug in cfn-hup?

File: cookbooks/aws-parallelcluster-install/templates/default/base/cfn-hup-runner.sh.erb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+# This script is used to run cfn-hup is a way that is suitable to be managed by supervisord.
+# In particular, cfn-hup is executed in no-daemon mode.
+#
+# Usage:   ./cfn-hup-runner.sh
+
+[ -f /etc/profile.d/proxy.sh ] && . /etc/profile.d/proxy.sh
+
+while true; do
+  <%= node['cluster']['cfn_bootstrap_virtualenv_path'] %>/bin/cfn-hup --no-daemon --verbose
```
Comment (enrico-usai) [inline cookbooks/aws-parallelcluster-install/templates/default/base/cfn-hup-runner.sh.erb:12]: If I'm not wrong cfn-hup log messages are going to cfn* log files. Can you check if cfn logs are correctly populated (and uploaded on CloudWatch) after the restart and update of the cluster?

File: CHANGELOG.md (hunk 1 above)
Comment (gmarciani) [reply CHANGELOG.md:34]: If cfn-hup is run in daemon mode and it is then killed, the following daemons restarted by supervisord systematically fail after ~min due to a dangling lock on the PID file. So if the cfn-hup daemon is killed, then the cluster will never be able to perform a cluster update.

I'll open an issue to the CloudFormation team on this.

File: cookbooks/aws-parallelcluster-install/templates/default/base/cfn-hup-runner.sh.erb (hunk 2 above)
Comment (gmarciani) [reply cookbooks/aws-parallelcluster-install/templates/default/base/cfn-hup-runner.sh.erb:12]: Confirmed: cfn logs are populated and sent to CloudWatch. 
Just to be precise: cfn-hup logs are currently not sent to CW, but only cloud-init ones

**Review by enrico-usai — APPROVED** [review]

**Pull request merged**